apps/mobile/src/features/send-refactor/utils.ts

```diff
@@ -0,0 +1,124 @@
+import { type BtcFormSchema } from '@/features/send-refactor/forms/btc/btc-form-schema';
+import { type StxFormSchema } from '@/features/send-refactor/forms/stx/stx-form-schema';
+import { type Account } from '@/store/accounts/accounts';
+import { bytesToHex } from '@noble/hashes/utils';
+import { type PublicKey } from '@stacks/common';
+import { ChainId, type StacksNetwork } from '@stacks/network';
+import BigNumber from 'bignumber.js';
+import memoize from 'p-memoize';
+
+import {
+  type BitcoinNativeSegwitPayer,
+  type CoinSelectionUtxo,
+  createBitcoinAddress,
+  generateBitcoinUnsignedTransactionNativeSegwit,
+  getBtcSignerLibNetworkConfigByMode,
+  isValidBitcoinAddress,
+  payerToBip32Derivation,
+} from '@leather.io/bitcoin';
+import { STX_DECIMALS } from '@leather.io/constants';
+import { type BitcoinNetworkModes, type Money } from '@leather.io/models';
+import { type Utxo, defaultStacksFees, isAddressCompliant } from '@leather.io/query';
+import {
+  TransactionTypes,
+  generateStacksUnsignedTransaction,
+  isValidStacksAddress,
+} from '@leather.io/stacks';
+import {
+  convertAmountToBaseUnit,
+  createMoney,
+  createMoneyFromDecimal,
+  stxToMicroStx,
+} from '@leather.io/utils';
+
+export function calculateDefaultStacksFee() {
+  const defaultFeeFallback = 2500;
+  const defaultFeeFallbackAsMoney = createMoney(defaultFeeFallback, 'STX');
+  return convertAmountToBaseUnit(defaultStacksFees.estimates[1]?.fee ?? defaultFeeFallbackAsMoney);
+}
+
+export function calculateStxMaxSpend(availableBalance: Money, fee: BigNumber) {
+  return convertAmountToBaseUnit(availableBalance.amount.minus(stxToMicroStx(fee)), STX_DECIMALS);
+}
+
+export function createCoinSelectionUtxos(utxos: Utxo[]): CoinSelectionUtxo[] {
+  return utxos.map(utxo => ({
+    address: utxo.address,
+    txid: utxo.txid,
+    value: Number(utxo.value),
+    vout: utxo.vout,
+  }));
+}
+
+export function isUserInputEffectivelyZero(input: string) {
+  return input === '0' || /^0\.0*$/.test(input);
+}
+
+export function getInitialRelevantAccounts(accounts: Account[]) {
+  return accounts.filter(account => account.status !== 'hidden');
+}
+
+export function validateDecimalPlaces(value: string, maxDecimalPlaces: number) {
+  return (value.split('.')[1] ?? '').length <= maxDecimalPlaces;
+}
```

Comment:
unit tests?

Reply:
Can you please add tests for these functions?